lib/commons/table/get-headers.js

```diff
@@ -15,13 +15,21 @@ import findUp from '../dom/find-up';
 function traverseForHeaders(headerType, position, tableGrid) {
 	const property = headerType === 'row' ? '_rowHeaders' : '_colHeaders';
 	const predicate = headerType === 'row' ? isRowHeader : isColumnHeader;
+	const startCell = tableGrid[position.y][position.x];
+
+	// adjust position by rowspan and colspan
+	const colspan = parseInt(startCell.getAttribute('colspan') || 0);
+	const rowspan = parseInt(startCell.getAttribute('rowspan') || 0);
+	const rowStart = position.y + rowspan;
+	const colStart = position.x + colspan;
```

Comment:
This is going to break if someone passes a negative span. `<td rowspan="-5">` might not do anything, but someone could still set it.